### ISSUE TITLE
Feature Request:  be able to reuse filter states for filtering data in batches, or have ability to initialize filter states for IIR filters

### ISSUE THREAD

**Issue by dph10** [psambit9791/jDSP#34]
As currently written  and as an example, when we call Butterworth.lowPassFilter() method a new filter is initialized and used to filter the data.  This works correctly and as expected when all data is already available ahead of time.

In same cases (using a continuous sampling data acquisition task) we may  want to collect a batch of data, filter that batch, and then display the filtered data.  We may want to do this over and over again as new batches of data is collected.  If we use the filters in this cases, a new filter is created and initialized each time we call the .lowPassFilter() method, which will incorrectly show discontinuous jumps in the filtered and displayed data from batch to batch.

I am proposing to have a "BatchFilter" or reusable filter, where the same filter is persisted and reused, as opposed to creating a new filter for every filter operation.  I think this should hopefully be straightforward, as you could just store uk.me.berndporr.iirj.Butterworth() filter as a field in your ButterWorth class and reused that filter for each filter operation, instead of initializing and using a new one with each filter operation.

**Comment by psambit9791** [comment]
@dph10 Could you please add an example usage so that I can replicate this problem?

**Comment by psambit9791** [comment]
Closing due to lack of clarification

**Comment by psambit9791** [comment]
Based on some code review, I understand the issue was primarily arising from the need to initialise the filters with the signals.

For example, for **Butterworth filter**, we were using the following:
```
Butterworth flt = new Butterworth(signal, Fs); //signal is of type double[]  
double[] result = flt.lowPassFilter(order, cutOff);
```

With the latest update, this has been changed to allow for the following:
```
Butterworth flt = new Butterworth(Fs);
double[] result = flt.lowPassFilter(signal, order, cutOff); //signal is of type double[]    
```

This allows the filter object to be reused with different signals.

To be released with v2.0.0.
